sentinel/test/mocha/lib/message-utils.js

```diff
@@ -0,0 +1,331 @@
+          .should.equal(clinicPhone);
+      });
+      it('resolves parent correctly', () => {
+        utils._getRecipient(doc, 'parent')
+          .should.equal(parentPhone);
+      });
+      it('resolves grandparent correctly', () => {
+        utils._getRecipient(doc, 'grandparent')
+          .should.equal(grandparentPhone);
+      });
+    });
+    it('tries to resolve the value from the fields property', () => {
+      utils._getRecipient({fields: {foo: 'bar'}}, 'foo')
+        .should.equal('bar');
+    });
+    it('tries to resolve simple values directly on the doc', () => {
+      utils._getRecipient({foo: 'bar'}, 'foo')
+        .should.equal('bar');
+    });
+    it('tries to resolve complex values directly on the doc', () => {
+      utils._getRecipient({foo: {bar: {smang: 'baz'}}}, 'foo.bar.smang')
+        .should.equal('baz');
+    });
+    it('is doc.from if the recipient cannot be resolved', () => {
+      utils._getRecipient({from: 'foo'}, 'a-recipient')
+        .should.equal('foo');
+    });
+  });
+
+  describe('generate', () => {
+
+    it('adds uuid', done => {
+      sinon.stub(uuid, 'v4').returns('some-uuid');
+      const config = {};
+      const translate = null;
+      const doc = {};
+      const content = { message: 'xxx' };
+      const recipient = '+1234';
+      const messages = utils.generate(config, translate, doc, content, recipient);
+      expect(messages.length).to.equal(1);
+      const message = messages[0];
+      expect(message.message).to.equal('xxx');
+      expect(message.to).to.equal('+1234');
+      expect(message.uuid).to.equal('some-uuid');
+      done();
+    });
+
+    describe('truncation', () => {
+
+      it('does not truncate short sms', done => {
+        const sms = generateMessage(MAX_GSM_LENGTH);
+        const config = { multipart_sms_limit: 10 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('does not truncate short unicode sms', done => {
+        const sms = generateMessage(MAX_UNICODE_LENGTH, true);
+        const config = { multipart_sms_limit: 10 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('truncates long sms', done => {
+        const sms = generateMessage(1000);
+        const expected = sms.substr(0, 150) + '...';
```

Comment:
Why does this truncate at 150 characters?  If the sms part limit is 1, it could do 157 chars + `...`.

Reply:
We use the gsm library to handle this for us and it [looks like](https://github.com/vchatterji/gsm/blob/master/lib/gsm.js#L44) they allow a single message to be 160 but a split message can only be 153. I can't find any documentation on why this might be but perhaps is to allow for any special characters that indicate there are more messages to come. It might even be carrier specific so our code is just a guess as to how many SMS the message will result in.